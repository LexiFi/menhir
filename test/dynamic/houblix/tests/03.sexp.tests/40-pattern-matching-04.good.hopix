type option = None | Some (int)

let x : int = switch (Some<>(42)) { Some<>((x : int)) -> x | None<> -> 0 }