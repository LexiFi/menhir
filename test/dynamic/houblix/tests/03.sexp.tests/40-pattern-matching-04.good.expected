(((value
    (DefineType
      ((value (TCon option))
        (position ((start_p ("" 1 0 5)) (end_p ("" 1 0 11)))))
      ()
      (DefineSumType
        ((((value (KId None))
            (position ((start_p ("" 1 0 14)) (end_p ("" 1 0 18)))))
           ())
          (((value (KId Some))
             (position ((start_p ("" 1 0 21)) (end_p ("" 1 0 25)))))
            (((value (TyCon (TCon int) ()))
               (position ((start_p ("" 1 0 27)) (end_p ("" 1 0 30)))))))))))
   (position ((start_p ("" 1 0 0)) (end_p ("" 1 0 31)))))
  ((value
     (DefineValue
       (SimpleValue
         (((value (Id x))
            (position ((start_p ("" 3 33 37)) (end_p ("" 3 33 38)))))
           (((value
               (ForallTy ()
                 ((value (TyCon (TCon int) ()))
                   (position ((start_p ("" 3 33 41)) (end_p ("" 3 33 44)))))))
              (position ((start_p ("" 3 33 40)) (end_p ("" 3 33 44))))))
           ((value
              (Case
                ((value
                   (Tagged
                     ((value (KId Some))
                       (position
                         ((start_p ("" 3 33 55)) (end_p ("" 3 33 59)))))
                     (())
                     (((value
                         (Literal
                           ((value (LInt 42))
                             (position
                               ((start_p ("" 3 33 62)) (end_p ("" 3 33 64)))))))
                        (position
                          ((start_p ("" 3 33 62)) (end_p ("" 3 33 64))))))))
                  (position ((start_p ("" 3 33 55)) (end_p ("" 3 33 65)))))
                (((value
                    (Branch
                      ((value
                         (PTaggedValue
                           ((value (KId Some))
                             (position
                               ((start_p ("" 3 33 69)) (end_p ("" 3 33 73)))))
                           (())
                           (((value
                               (PTypeAnnotation
                                 ((value
                                    (PVariable
                                      ((value (Id x))
                                        (position
                                          ((start_p ("" 3 33 77))
                                            (end_p ("" 3 33 78)))))))
                                   (position
                                     ((start_p ("" 3 33 77))
                                       (end_p ("" 3 33 78)))))
                                 ((value (TyCon (TCon int) ()))
                                   (position
                                     ((start_p ("" 3 33 81))
                                       (end_p ("" 3 33 84)))))))
                              (position
                                ((start_p ("" 3 33 76)) (end_p ("" 3 33 85))))))))
                        (position
                          ((start_p ("" 3 33 69)) (end_p ("" 3 33 86)))))
                      ((value
                         (Variable
                           ((value (Id x))
                             (position
                               ((start_p ("" 3 33 90)) (end_p ("" 3 33 91)))))
                           ()))
                        (position
                          ((start_p ("" 3 33 90)) (end_p ("" 3 33 91)))))))
                   (position ((start_p ("" 3 33 69)) (end_p ("" 3 33 91)))))
                  ((value
                     (Branch
                       ((value
                          (PTaggedValue
                            ((value (KId None))
                              (position
                                ((start_p ("" 3 33 94)) (end_p ("" 3 33 98)))))
                            (()) ()))
                         (position
                           ((start_p ("" 3 33 94)) (end_p ("" 3 33 100)))))
                       ((value
                          (Literal
                            ((value (LInt 0))
                              (position
                                ((start_p ("" 3 33 104))
                                  (end_p ("" 3 33 105)))))))
                         (position
                           ((start_p ("" 3 33 104)) (end_p ("" 3 33 105)))))))
                    (position ((start_p ("" 3 33 94)) (end_p ("" 3 33 105))))))))
             (position ((start_p ("" 3 33 47)) (end_p ("" 3 33 107)))))))))
    (position ((start_p ("" 3 33 33)) (end_p ("" 3 33 107))))))
